### DISCUSSION TITLE
Some observations during upgrading from 1.12 to 2.2

### DISCUSSION THREAD

**Discussion by GinTonic99** [gerbera/gerbera#3190] [General]
Hi,
since more than 15 years I am running  first mediatomb now gerbera to handle my music, photos and videos. Since the beginning I am using my own import scripts and I am running two instances of gerbera one for music and the other one for photos and videos.
Last weekend I upgraded the music server from 1.12 to 2.2. It took me some time and various attempts until I understood the new concept  but when it started working, I was positively surprised, that my import script was working. However I noticed that some of the Albums appeared more than once under the same name in the tree with the tracks distributed among the instances of the same album. My investigations showed that this happens in the following cases:

- different entries in the Date tags of the tracks. As I do not use the date tag, I either deleted them or unified the entries. I have many albums that are theme oriented, samplers, or best of albums with tracks from different years. For such albums, this concept simply does not work
- different entries in the AlbumArtist tags. ok, this should not be.  As I do not use this tag, I either deleted them or unified the entries.
- Different entries in tags like Remark or AlbumArt. I did not investigate further but only deleted the concerned tags.
- still there were some albums remaining where I did not find a difference in the tag contents, however my suspicion was, that the tags of some tracks had been updated later than others. I therfore overwrote all tags of the concerned albums, this repaired the issue. My suspicion is, that different character sets have been involved.

It was quite a surprise for me, to find such hidden automatisms being active even when I use my own import script. I would expect that as per the imput script, the albums should be identified by their name only an not by some hidden algorithm under the hood.
As detailed above, I found workarouds that I can live with. I write this just in case somebody else is stepping across and puzzled by similar effects.

One last question: in 1.12 I managed to suppress the display of album art in the tree view of hte browser. I did not find a setting for this in 2.2. Did I overlook this?

Thanks for gerbera
best regards
Gin

**Comment by KarlStraussberger** [comment]
I am really impressed to see that only one configuration entries as such an impact 😄 

Sorry that we changed the javascript API after 15 years. Luckily you waited until 2.2 because there several small changes in between.

The thing behind all this is the new setting https://docs.gerbera.io/en/latest/config-import.html#virtual-directories which controls the virtual path we create internally to do exactly what you observed: Separate items that are not the same! For this to work correctly the metadata must be in good shape which you now did establish.

The thumbnails should still be in https://docs.gerbera.io/en/latest/config-server.html#ui (show-thumbnail) if there is no hidden bug.

**Reply by GinTonic99** [reply]
Thank you for the response and the hints. 
If I understand the idea of the virtual-directories correctly, they should only be avtive with the respective configuration option is present in the configuration file. This is, however , not the case for me, I don't have the virtual-directories section in my configuration file. 
BTW  I get numerous warnings like this below:
`Error: Failed to open /Album/-- Alle --/Tschaikowsky: Balletsuiten (Karajan)@2024: Datei oder Verzeichnis nicht gefunden
`
The show-thumbnail option is working, I did not see it. Thanks for this hint as well.

**Reply by KarlStraussberger** [reply]
`virtual-directories` defaults to "M_ALBUMARTIST", "M_UPNP_DATE" which explains the exact problems you had. (small update to the docs helps)

The error message is quite suspicious but may be triggered by your import script (this is the internal location string for virtual entries): Check the properties you set!

**Comment by GinTonic99** [comment]
When I opened this thread, I was still playing with the new version on my notebook. During the past days I have upgraded Gerbera on my server. The setup is exactly the same as with 1.12, same machine, same OS, same music collection. I am using mysql as database and persistive timed scan. Before upgrading, I scrapped the old database in order to avoid upgrade issues. 

Below the observations with this setup:

- still some albums are split and I cannot find out why. As they are few, I can live with it, however an option to disable this album split automatism would be appreciated.
- the timed scanning (after the database is already completely filled) is significantly slower as it was with 1.12. It took 1-2 hours before, now it takes more than 6 hours. The mariadb process is running constantly with 95-100% CPU load during scan.
- During scan, the server is very irresponsive, with a delay in the range of a minute when browsing through the music tree. This was not the case with 1.12. Playing music does work however.
- I took  the date related entry out of the import script and the error messages mentioned above seem to have disappeared, thanks for the hint

I can live with all mentioned above, an improvement regarding scan time and responsiveness would however be appreciated.

Thank you for your work and the continous support.

**Reply by KarlStraussberger** [reply]
It's hard to address performance issues remotely. In fact we removed serveral SELECT calls to the database and cache the objects in main memory.
Some questions that might help:
- Are database and media files on same disk?
- Is the database in some direct write mode?
- Is there some main memory left when gerbera and maria do their work?

For the album split you can try the new option `<import case-sensitive-tags="no" import-mode="grb">` to lowercase all metadata in virtual path (only visible in the new (two PRs ago) tooltip on web UI).
The `import-mode` may also help to speed up the import, it's still new but stable on my system for the last year.

**Reply by GinTonic99** [reply]
thanks for the response,
to answer your questions:

    _Are database and media files on same disk?_
no, gerbera is running in a virtualbox virtual machine and the data is on a raid on the host, acces via NFS. I know this setup is slow. However I have this setup since 10 years and only now with this last upgrade I noticed such a significant perfomance drop. While I dont mind that scanning takes long time, the irresponsivesness during a scan of gerbera is annoying now. As a workaround I have now set a scan interval on one week, I can increase it more as next step. The fact that mariadb consumes almost 100% of CPU load without any wait perecentage indicated in top during a scan seems to indicate that the bottleneck is directly in the database and not in the data access.

    _Is the database in some direct write mode?_
I dont know and would not now how to find out, the SQL server is running on the same virtual machine as gerbera and I am not aware that I changed any configuration. Going through the config files I did not step over a suspective entry.

    _Is there some main memory left when gerbera and maria do their work?_
yes, I have more than 1GB of free memory all the time

I will not try your suggestion with the import-case as I can live with the few split albums.
I did not yet test your suggestions about import-mode but willl try at next occasion.

thanks again br Gin
